cpp/src/arrow/util/io_util.cc

```diff
@@ -979,6 +979,32 @@ static Status StatusFromMmapErrno(const char* prefix) {
   return IOErrorFromErrno(errno, prefix);
 }
 
+namespace {
+
+int64_t GetPageSizeInternal() {
+#if defined(__APPLE__)
+  return getpagesize();
+#elif defined(_WIN32)
+  SYSTEM_INFO si;
+  GetSystemInfo(&si);
+  return si.dwPageSize;
+#else
+  errno = 0;
+  const auto ret = sysconf(_SC_PAGESIZE);
+  if (ret == -1) {
+    ARROW_LOG(FATAL) << "sysconf(_SC_PAGESIZE) failed: " << ErrnoMessage(errno);
```

Comment:
Just return 4k?

Reply:
I'd rather know in which conditions this syscall fails (it should never).